cirq-google/cirq_google/experimental/noise_models/fidelity.py

```diff
@@ -0,0 +1,124 @@
+import cirq
+import numpy as np
+from typing import Optional, Sequence
+from cirq.devices.noise_model import validate_all_measurements
+from itertools import product
+
+class Fidelity():
```

Comment:
Use of a `dataclass` might make life a little easier here.

Reply:
I looked into this, do you think it would still be worthwhile since I have to do validate the parameters in the init function?